drivers/dac/dac_dacx0508.c

```diff
@@ -0,0 +1,489 @@
+	if (ret < 0) {
+		return ret;
+	}
+	tmp = (regval[0] << 8) | regval[1];
+
+	if (setting) {
+		tmp |= mask;
+	} else {
+		tmp &= ~mask;
+	}
+
+	regval[0] = tmp >> 8;
+	regval[1] = tmp & 0xFF;
+
+	ret = dacx0508_reg_write(dev, addr, regval);
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int dacx0508_channel_setup(struct device *dev,
+			    const struct dac_channel_cfg *channel_cfg)
+{
+	const struct dacx0508_config *config = dev->config_info;
+	struct dacx0508_data *data = dev->driver_data;
+
+	if (channel_cfg->channel_id > DACX0508_MAX_CHANNEL - 1) {
+		LOG_ERR("Unsupported channel %d", channel_cfg->channel_id);
+		return -ENOTSUP;
+	}
+
+	if (channel_cfg->resolution != config->resolution) {
+		LOG_ERR("Unsupported resolution %d", channel_cfg->resolution);
+		return -ENOTSUP;
+	}
+
+	data->configured |= BIT(channel_cfg->channel_id);
+
+	return 0;
+}
+
+static int dacx0508_write_value(struct device *dev, uint8_t channel,
+				uint32_t value)
+{
+	const struct dacx0508_config *config = dev->config_info;
+	struct dacx0508_data *data = dev->driver_data;
+	uint8_t regval[2];
+	int ret;
+
+	if (channel > DACX0508_MAX_CHANNEL - 1) {
+		LOG_ERR("unsupported channel %d", channel);
+		return -ENOTSUP;
+	}
+
+	if (!(data->configured & BIT(channel))) {
+		LOG_ERR("Channel not initialized");
+		return -EINVAL;
+	}
+
+	if (value >= (1 << config->resolution)) {
+		LOG_ERR("Value %d out of range", value);
+		return -EINVAL;
+	}
+
+	value <<= (16 - config->resolution);
+	regval[0] = value >> 8;
+	regval[1] = value & 0xFF;
+
+	ret = dacx0508_reg_write(dev, DACX0508_REG_DAC0 + channel, regval);
+	if (ret) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int dacx0508_soft_reset(struct device *dev)
+{
+	uint8_t regval[2] = {0, DACX0508_MASK_TRIGGER_SOFT_RESET};
+	int ret;
+
+	ret = dacx0508_reg_write(dev, DACX0508_REG_TRIGGER, regval);
+	if (ret) {
+		return -EIO;
+	}
+	k_usleep(DACX0508_POR_DELAY);
+
+	return 0;
+}
+
+static int dacx0508_device_id_check(struct device *dev)
+{
+	const struct dacx0508_config *config = dev->config_info;
+	uint8_t regval[2] = {0, };
+	uint8_t resolution;
+	uint16_t dev_id;
+	int ret;
+
+	ret = dacx0508_reg_read(dev, DACX0508_REG_DEVICE_ID, regval);
+	if (ret) {
+		LOG_ERR("Unable to read Device ID");
+		return -EIO;
+	}
+	dev_id = (regval[0] << 8) | regval[1];
+
+	resolution = dev_id >> 12;
+	if (resolution != (16 - config->resolution) >> 1) {
+		LOG_ERR("Not match chip resolution");
+		return -EINVAL;
+	}
+
+	if ((dev_id & DACX0508_MASK_DEVICE_ID_8CH) !=
+				DACX0508_MASK_DEVICE_ID_8CH) {
+		LOG_ERR("Support channels mismatch");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int dacx0508_setup(struct device *dev)
+{
+	const struct dacx0508_config *config = dev->config_info;
+	uint8_t regval[2] = {0, };
+	uint8_t tmp = 0;
+	int ret;
+
+	switch (config->reference) {
+	case DAC_REF_INTERNAL_1:
+		ret = dacx0508_reg_update(dev, DACX0508_REG_CONFIG,
+				DACX0508_MASK_CONFIG_REF_PWDWN, false);
```

Comment:
Please indent these continued function call lines according to the coding style (tabs followed by spaces until they align with the first argument on the line above).

You could use the switch statement to merely store the values to be written to the config and gain registers and have only two calls to `dacx0508_reg_update()` after the switch. That would make the code more readable, I think.